userspace/libsinsp/test/public_sinsp_API/interesting_syscalls.cpp

```diff
@@ -0,0 +1,435 @@
+
+#ifdef __NR_capset
+	PPM_SC_CAPSET,
+#endif
+
+#ifdef __NR_chdir
+	PPM_SC_CHDIR,
+#endif
+
+#ifdef __NR_chroot
+	PPM_SC_CHROOT,
+#endif
+
+#ifdef __NR_clone
+	PPM_SC_CLONE,
+#endif
+
+#ifdef __NR_clone3
+	PPM_SC_CLONE3,
+#endif
+
+#ifdef __NR_close
+	PPM_SC_CLOSE,
+#endif
+
+#ifdef __NR_connect
+	PPM_SC_CONNECT,
+#endif
+
+#ifdef __NR_creat
+	PPM_SC_CREAT,
+#endif
+
+#ifdef __NR_dup
+	PPM_SC_DUP,
+#endif
+
+#ifdef __NR_dup2
+	PPM_SC_DUP2,
+#endif
+
+#ifdef __NR_dup3
+	PPM_SC_DUP3,
+#endif
+
+#ifdef __NR_eventfd
+	PPM_SC_EVENTFD,
+#endif
+
+#ifdef __NR_eventfd2
+	PPM_SC_EVENTFD2,
+#endif
+
+#ifdef __NR_execve
+	PPM_SC_EXECVE,
+#endif
+
+#ifdef __NR_execveat
+	PPM_SC_EXECVEAT,
+#endif
+
+#ifdef __NR_fchdir
+	PPM_SC_FCHDIR,
+#endif
+
+#ifdef __NR_fcntl
+	PPM_SC_FCNTL,
+#endif
+
+#ifdef __NR_fcntl64
+	PPM_SC_FCNTL64,
+#endif
+
+#ifdef __NR_fork
+	PPM_SC_FORK,
+#endif
+
+#ifdef __NR_inotify_init
+	PPM_SC_INOTIFY_INIT,
+#endif
+
+#ifdef __NR_inotify_init1
+	PPM_SC_INOTIFY_INIT1,
+#endif
+
+#ifdef __NR_io_uring_setup
+	PPM_SC_IO_URING_SETUP,
+#endif
+
+#ifdef __NR_mount
+	PPM_SC_MOUNT,
+#endif
+
+#ifdef __NR_open
+	PPM_SC_OPEN,
+#endif
+
+#ifdef __NR_open_by_handle_at
+	PPM_SC_OPEN_BY_HANDLE_AT,
+#endif
+
+#ifdef __NR_openat
+	PPM_SC_OPENAT,
+#endif
+
+#ifdef __NR_openat2
+	PPM_SC_OPENAT2,
+#endif
+
+#ifdef __NR_pipe
+	PPM_SC_PIPE,
+#endif
+
+#ifdef __NR_pipe2
+	PPM_SC_PIPE2,
+#endif
+
+#ifdef __NR_prlimit64
+	PPM_SC_PRLIMIT64,
+#endif
+
+#ifdef __NR_recvfrom
+	PPM_SC_RECVFROM,
+#endif
+
+#ifdef __NR_recvmsg
+	PPM_SC_RECVMSG,
+#endif
+
+#ifdef __NR_getsockopt
+	PPM_SC_GETSOCKOPT, /// TODO: In the next future probably we could remove this from the state
+#endif
+
+#ifdef __NR_sendmsg
+	PPM_SC_SENDMSG,
+#endif
+
+#ifdef __NR_sendto
+	PPM_SC_SENDTO,
+#endif
+
+#ifdef __NR_setgid
+	PPM_SC_SETGID,
+#endif
+
+#ifdef __NR_setgid32
+	PPM_SC_SETGID32,
+#endif
+
+#ifdef __NR_setpgid
+	PPM_SC_SETPGID,
+#endif
+
+#ifdef __NR_setresgid
+	PPM_SC_SETRESGID,
+#endif
+
+#ifdef __NR_setresgid32
+	PPM_SC_SETRESGID32,
+#endif
+
+#ifdef __NR_setresuid
+	PPM_SC_SETRESUID,
+#endif
+
+#ifdef __NR_setresuid32
+	PPM_SC_SETRESUID32,
+#endif
+
+#ifdef __NR_setrlimit
+	PPM_SC_SETRLIMIT,
+#endif
+
+#ifdef __NR_setsid
+	PPM_SC_SETSID,
+#endif
+
+#ifdef __NR_setuid
+	PPM_SC_SETUID,
+#endif
+
+#ifdef __NR_setuid32
+	PPM_SC_SETUID32,
+#endif
+
+#ifdef __NR_shutdown
+	PPM_SC_SHUTDOWN,
+#endif
+
+#ifdef __NR_signalfd
+	PPM_SC_SIGNALFD,
+#endif
+
+#ifdef __NR_signalfd4
+	PPM_SC_SIGNALFD4,
+#endif
+
+#ifdef __NR_socket
+	PPM_SC_SOCKET,
+#endif
+
+#ifdef __NR_socketpair
+	PPM_SC_SOCKETPAIR,
+#endif
+
+#ifdef __NR_timerfd_create
+	PPM_SC_TIMERFD_CREATE,
+#endif
+
+#ifdef __NR_umount2
+	PPM_SC_UMOUNT2,
+#endif
+
+#ifdef __NR_userfaultfd
+	PPM_SC_USERFAULTFD,
+#endif
+
+#ifdef __NR_vfork
+	PPM_SC_VFORK,
+#endif
+};
+
+/* This test asserts that `enforce_sinsp_state_ppm_sc` correctly retrieves
+ * the `libsinsp` state ppm_sc set.
+ */
+TEST(InterestingSyscalls, enforce_sinsp_state_basic)
```

Comment:
nit: I think all other tests use the `snake_case` convention for both the test type and name

Reply:
yeah i will change it :) ty